include/ebpf_store_helper.h

```diff
@@ -3,83 +3,55 @@
 
 #pragma once
 
+#ifndef USER_MODE
+#include "framework.h"
+#else
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <winreg.h>
+#endif
+
 #include "ebpf_program_types.h"
-#include "ebpf_registry_helper.h"
 #include "ebpf_windows.h"
 
-#ifdef USER_MODE
-extern ebpf_registry_key_t ebpf_root_registry_key;
-#endif
+#define GUID_STRING_LENGTH 38 // not including the null terminator.
 
-static uint32_t
-_ebpf_store_open_or_create_provider_registry_key(_Out_ ebpf_registry_key_t* provider_key)
-{
-    __return_type status = _SUCCESS;
-    ebpf_registry_key_t root_key = NULL;
-    *provider_key = NULL;
-
-    // Open (or create) root eBPF registry path.
 #ifndef USER_MODE
-    status = create_registry_key(NULL, EBPF_ROOT_REGISTRY_PATH, REG_CREATE_FLAGS, &root_key);
+#define __return_type NTSTATUS
+#define _SUCCESS STATUS_SUCCESS
+#define IS_SUCCESS(x) (NT_SUCCESS(x))
+typedef HANDLE ebpf_registry_key_t;
+typedef _Return_type_success_(NT_SUCCESS(return )) uint32_t ebpf_registry_result_t;
 #else
-    status = create_registry_key(ebpf_root_registry_key, EBPF_ROOT_RELATIVE_PATH, REG_CREATE_FLAGS, &root_key);
+#define __return_type uint32_t
+#define IS_SUCCESS(x) (x == ERROR_SUCCESS)
```

Comment:
Why two different return types rather than picking one and converting the other to it? (since we already have conversion functions)

Reply:
For return values of functions we implement, we usually use ebpf_result_t for this purpose.  Can we do that?

Reply:
> Why two different return types rather than picking one and converting the other to it? (since we already have conversion functions)

@dthaler can you please clarify what conversion functions are you referring to?

Reply:
@dthaler @saxena-anurag For registry functions, we also have `ebpf_registry_result_t`... how do we want to unify?

Reply:
ebpf_result_t

Reply:
@dthaler what conversion functions were you thinking about? 

For KM, the required conversion function is in a separate library, which would need to be linked (and increase the store lib), i.e.: `ntstatus_to_ebpf_result`; but this is only defined in `ebpf_platform_user.cpp`, whereas we'd eventually need it in a KM library. In fact, in its implementation, `ntstatus_to_ebpf_result` calls `_ntstatus_to_win32_error_code`, which in turn loads `ntdll.dll` to decode the error codes.

Reply:
Right now the precedent (for good or bad) is for each function to check NT_SUCCESS() and do the conversion itself even if to just map all failures to EBPF_FAILED.  See ebpf_update_global_helpers() for instance and other functions in ebpf_platform_kernel.c.